content/developer/howtos/webservices.rst

```diff
@@ -0,0 +1,150 @@
+============
+Web Services
+============
+
+The web-service module offer a common interface for all web-services :
+
+- XML-RPC
+- JSON-RPC
+
+Business objects can also be accessed via the distributed object
+mechanism. They can all be modified via the client interface with contextual
+views.
+
+Odoo is accessible through XML-RPC/JSON-RPC interfaces, for which libraries
+exist in many languages.
+
+XML-RPC Library
+---------------
+
+The following example is a Python 3 program that interacts with an Odoo
+server with the library ``xmlrpc.client``::
+
+   import xmlrpc.client
+
+   root = 'http://%s:%d/xmlrpc/' % (HOST, PORT)
+
+   uid = xmlrpc.client.ServerProxy(root + 'common').login(DB, USER, PASS)
+   print("Logged in as %s (uid: %d)" % (USER, uid))
+
+   # Create a new note
+   sock = xmlrpc.client.ServerProxy(root + 'object')
+   args = {
+       'color' : 8,
+       'memo' : 'This is a note',
+       'create_uid': uid,
+   }
+   note_id = sock.execute(DB, uid, PASS, 'note.note', 'create', args)
+
+.. exercise:: Add a new service to the client
+
+   Write a Python program able to send XML-RPC requests to a PC running
+   Odoo (yours, or your instructor's). This program should display all
+   the sessions, and their corresponding number of seats. It should also
+   create a new session for one of the courses.
+
+   .. only:: solutions
+
+        .. code-block:: python
+
+            import functools
+            import xmlrpc.client
+            HOST = 'localhost'
+            PORT = 8069
+            DB = 'openacademy'
+            USER = 'admin'
+            PASS = 'admin'
+            ROOT = 'http://%s:%d/xmlrpc/' % (HOST,PORT)
+
+            # 1. Login
+            uid = xmlrpc.client.ServerProxy(ROOT + 'common').login(DB,USER,PASS)
+            print("Logged in as %s (uid:%d)" % (USER,uid))
+
+            call = functools.partial(
+                xmlrpc.client.ServerProxy(ROOT + 'object').execute,
+                DB, uid, PASS)
+
+            # 2. Read the sessions
+            sessions = call('openacademy.session','search_read', [], ['name','seats'])
+            for session in sessions:
+                print("Session %s (%s seats)" % (session['name'], session['seats']))
+            # 3.create a new session
+            session_id = call('openacademy.session', 'create', {
+                'name' : 'My session',
+                'course_id' : 2,
+            })
+
+        Instead of using a hard-coded course id, the code can look up a course
+        by name::
+
+            # 3.create a new session for the "Functional" course
+            course_id = call('openacademy.course', 'search', [('name','ilike','Functional')])[0]
+            session_id = call('openacademy.session', 'create', {
+                'name' : 'My session',
+                'course_id' : course_id,
+            })
+
+.. note::
+
+    There is an in-depth tutorial on XML-RPC with examples spanning over multiple programming
```

Comment:
```suggestion
    There is an in-depth tutorial on XML-RPC with examples spanning multiple programming
```
But this should rather be a `seealso` too. This is generally preferable when you mention the existence of another page that might interest the reader. Notes are more for additional comments on the content of the current page.
```rst
.. seealso::
   - :doc:`../api/external_api`: The in-depth tutorial on XML-RPC with examples spanning multiple programming languages.
```